timesketch/lib/analyzers/feature_extraction.py

```diff
@@ -146,6 +178,57 @@ def run(self):
 
         return ', '.join(return_strings)
 
+    @staticmethod
+    def _get_attribute_value(
+            current_val,
+            extracted_value,
+            keep_multi,
+            merge_values,
+            type_list):
+        """Return value to store.
+        Args:
+            current_val: current value of store_as.
+            extracted_value: values matched from regexp (type list).
+            keep_multi: choise if you keep all match from regex (type boolean).
+            merge_values: choise if you merge value from extracted
+                 and current (type boolean).
+            type_list: choise if you store values in list type(type boolean).
+        Returns:
+            Value to store
+        """
+        new_value = None
+        if not current_val:
+            merge_values = False
+        if len(extracted_value) == 1:
+            keep_multi = False
+        if type_list:
+            if merge_values:
+                if keep_multi:
+                    new_value = list(set(current_val) | set(extracted_value))
+                elif extracted_value[0] not in current_val:
+                    new_value = current_val + [extracted_value[0]]
+            else:
+                if keep_multi:
+                    new_value = extracted_value
+                else:
+                    new_value = [extracted_value[0]]
+        else:
+            if merge_values:
+                if keep_multi:
+                    add_list = []
+                    for elem_match in extracted_value:
+                        if elem_match not in current_val:
+                            add_list.append(elem_match)
```

Comment:
ok, so two things... if current_val is a comma separated string you can do:

```
current_list = [x.strip() fox x in current_val.split(',')]
new_set = set(current_list.extend(extracted_value))
return ','.join(list(new_set))
```

but if you just want to add this after the fact like you are doing here you can also do

```
extended_list = [x for x in extracted_value if x not in current_val]
extended_string = ','.join(extended_list)
return f'{current_val},{extended_string}'
```

since you are just extending teh list with the values that are not there